tests/attributes/test_attributes_scoped.py

```diff
@@ -584,4 +591,51 @@ def test_bulk_add_network_and_project_scoped_attribute(self, client, network_wit
         assert len(global_and_project_and_network_scoped_attributes) == len(all_global_attributes) + 4
 
 
+    def test_add_network_scoped_attribute_with_attribute_already_existing(self, client, network_with_data):
+        """
+            Test that when adding an attribute to a network, where that attribute is already
+            scoped higher, then return the higher scoped attribute instead of failing
+        """
+        global_attr = JSONObject({
+            "name": f'Global Attribute {datetime.datetime.now()}',
+            "dimension_id": None
+        })
+
+        network_scoped_attr = JSONObject({
+            "name": f'Network Attribute {datetime.datetime.now()}',
+            "dimension_id": None,
+            "network_id": network_with_data.id
+        })
+
+        project_attr_name =  f'Project Attribute {datetime.datetime.now()}'
+
+        project_scoped_attr = JSONObject({
+            "name": project_attr_name,
+            "dimension_id": None,
+            "project_id": network_with_data.project_id
+        })
+
+
+        previous_all_attributes = client.get_attributes()
+
+        new_attributes = client.add_attributes([global_attr, network_scoped_attr, project_scoped_attr])
+
+        scoped_ids = [a.id for a in new_attributes]
+
+        #Now add the attribute which is scoped to the project, to the network
+        duplicate_network_scoped_attr = JSONObject({
+            "name": project_attr_name,
+            "dimension_id": None,
+            "network_id": network_with_data.id
+        })
+
+
+        new_attributes = client.add_attributes([duplicate_network_scoped_attr])
+
+        #should return an existing scoped attribute (the project one)
+        assert new_attributes[0].id in scoped_ids
+
+        new_attribute = client.add_attribute(duplicate_network_scoped_attr)
 
+        #should return an existing scoped attribute (the project one)
+        assert new_attribute.id in scoped_ids
```

Comment:
Pedantic, but a stricter test would be that the attempt to add `duplicate_network_scoped_attr` via `add_attributes()` results in exactly one returned item which is the same `id` as that returned from 621 for the `project_attr_name` attr in particular: i.e. it found only one and it is the expected one.

Same for the 638 `add_attribute()` case, the returned `id` should be the specific one previously assigned to `project_attr_name` rather than any of the others in `scoped_ids`.